api/base/throttling.py

```diff
@@ -1,15 +1,40 @@
+from rest_framework.settings import api_settings as drf_settings
 from rest_framework.throttling import UserRateThrottle, AnonRateThrottle, SimpleRateThrottle
 
+from api.base import settings
+
 
 class BaseThrottle(SimpleRateThrottle):
 
     def failure(self, request):
         return False
 
+    def get_ident(self, request):
+        xff = request.META.get('HTTP_X_FORWARDED_FOR')
+        remote_addr = request.META.get('REMOTE_ADDR')
+        throttle_token = request.META.get('HTTP_X_THROTTLE_TOKEN')
+        num_proxies = drf_settings.NUM_PROXIES
+
+        if throttle_token:
+            return throttle_token
+
+        if num_proxies is not None:
+            if num_proxies == 0 or xff is None:
+                return remote_addr
+            addrs = xff.split(',')
+            client_addr = addrs[-min(num_proxies, len(addrs))]
+            return client_addr.strip()
+
+        return ''.join(xff.split()) if xff else remote_addr
+
     def allow_request(self, request, view):
         """
         Implement the check to see if the request should be throttled.
         """
+        throttle_token = settings.BYPASS_THROTTLE_TOKEN
```

Comment:
Superfluous var assignment -- the app should kill itself with the `assert` in `/api/settings/base/__init__.py` `if not settings.BYPASS_THROTTLE_TOKEN`. Just check `self.get_ident(request) == settings.BYPASS_THROTTLE_TOKEN` below.